Cannot read property 'startsWith' of undefined

friendly-errors-webpack-plugin:   "version": "1.7.0"
webpack:  "version": "4.20.2"


when code require a non-exists path like:
`
require.context('./components', false, /[\w-]+\.vue$/);
`

webpack will throw an exception:

`
ModuleNotFoundError: Module not found: Error: Can't resolve './components' in '/Users/unightsun/workspace/js/crawler-v2/cms-server/src'
    at factory.create (/Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/webpack/lib/Compilation.js:798:10)
    at asyncLib.parallel (/Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/webpack/lib/ContextModuleFactory.js:143:23)
    at /Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/neo-async/async.js:2817:7
    at /Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/neo-async/async.js:6783:13
    at contextResolver.resolve (/Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/webpack/lib/ContextModuleFactory.js:118:26)
    at doResolve (/Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/enhanced-resolve/lib/Resolver.js:184:12)
    at hook.callAsync (/Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/enhanced-resolve/lib/Resolver.js:238:5)
    at _fn0 (eval at create (/Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/tapable/lib/HookCodeFactory.js:32:10), <anonymous>:15:1)
    at resolver.doResolve (/Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:37:5)
    at hook.callAsync (/Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/enhanced-resolve/lib/Resolver.js:238:5)
    at _fn0 (eval at create (/Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/tapable/lib/HookCodeFactory.js:32:10), <anonymous>:15:1)
    at hook.callAsync (/Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/enhanced-resolve/lib/Resolver.js:238:5)
    at _fn0 (eval at create (/Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/tapable/lib/HookCodeFactory.js:32:10), <anonymous>:12:1)
    at resolver.doResolve (/Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:42:38)
    at hook.callAsync (/Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/enhanced-resolve/lib/Resolver.js:238:5)
    at _fn3 (eval at create (/Users/unightsun/workspace/js/crawler-v2/cms-server/node_modules/tapable/lib/HookCodeFactory.js:32:10), <anonymous>:42:1)
resolve './components' in '/Users/unightsun/workspace/js/crawler-v2/cms-server/src'
  using description file: /Users/unightsun/workspace/js/crawler-v2/cms-server/package.json (relative path: ./src)
    Field 'browser' doesn't contain a valid alias configuration
    using description file: /Users/unightsun/workspace/js/crawler-v2/cms-server/package.json (relative path: ./src/components)
      as directory
        /Users/unightsun/workspace/js/crawler-v2/cms-server/src/components doesn't exist
`

it's dependencies is:
`
[ RequireContextDependency {
    module: null,
    weak: false,
    optional: false,
    loc: SourceLocation { start: [Object], end: [Object] },
    options:
     { request: './components',
       recursive: false,
       regExp: /[\w-]+\.vue$/,
       mode: 'sync' },
    userRequest: './components',
    critical: false,
    hadGlobalOrStickyRegExp: false,
    range: [ 384, 438 ] } ]
`

so 'friendly-errors-webpack-plugin/src/transformers/moduleNotFound.js:17':
> const module = webpackError.dependencies[0].request;

will be undefined.

and then 'friendly-errors-webpack-plugin/src/formatters/moduleNotFound.js:5'
> return module.startsWith('./') || module.startsWith('../');

will throw a TypeError: Cannot read property 'startsWith' of undefined

optional fix will be 'friendly-errors-webpack-plugin/src/transformers/moduleNotFound.js:17':

> \- const module = webpackError.dependencies[0].request;
> \+ const dependency = webpackError.dependencies[0];
> \+ const module = dependency.request || dependency.options.request;

--- a/src/transformers/moduleNotFound.js
+++ b/src/transformers/moduleNotFound.js
@@ -13,7 +13,8 @@ function isModuleNotFoundError (e) {
 function transform(error) {
   const webpackError = error.webpackError;
   if (isModuleNotFoundError(error)) {
-    const module = webpackError.dependencies[0].request;
+    const dependency = webpackError.dependencies[0];
+    const module = dependency.request || dependency.options.request;
     return Object.assign({}, error, {
       message: `Module not found ${module}`,
       type: TYPE,

--- a/test/unit/transformers/moduleNotFound.spec.js
+++ b/test/unit/transformers/moduleNotFound.spec.js
@@ -1,5 +1,6 @@
 const moduleNotFound = require('../../../src/transformers/moduleNotFound');
 const expect = require('expect');
+const RequireContextDependency = require('webpack/lib/dependencies/RequireContextDependency');
 
 const error = {
   name: 'ModuleNotFoundError',
@@ -22,8 +23,7 @@ it('Sets the appropiate message', () => {
   expect(moduleNotFound(error).message).toEqual(message);
 });
 
-it('Sets the appropiate message', () => {
-  const message = 'Module not found redux';
+it('Sets the appropiate type', () => {
   expect(moduleNotFound({
     name: 'ModuleNotFoundError',
     message: 'Module not found',
@@ -34,4 +34,16 @@ it('Sets the appropiate message', () => {
 it('Ignores other errors', () => {
   const error = { name: 'OtherError' };
   expect(moduleNotFound(error)).toEqual(error);
+});
+
+it('Sets the correct message with a RequireContextDependency', () => {
+  const message = 'Module not found redux';
+
+  expect(moduleNotFound({
+      name: 'ModuleNotFoundError',
+      message: 'Module not found : redux',
+      webpackError: {
+        dependencies: [ new RequireContextDependency('redux') ],
+      },
+  }).message).toEqual(message);
 });

